backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +71,97 @@ def disambiguate_signatures(self, clusters):
+    matched_records = (
+        matched_record["inner_hits"]["authors"]["hits"]["hits"]
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    )
+    matched_authors_refs = set()
+    for record_authors in matched_records:
+        matched_authors_refs.update(
+            author["_source"]["record"]["$ref"]
+            for author in record_authors
+            if "record" in author["_source"]
+        )
+    return matched_authors_refs
+
+
+def assign_reference_to_author(matched_authors_refs, author):
+    if matched_authors_refs and len(matched_authors_refs) == 1:
+        author_reference_url = matched_authors_refs.pop()
+        author["record"] = {"$ref": author_reference_url}
+        return author_reference_url.split("/")[-1]
+
+
+def match_literature_author(author, updated_authors, record):
+    configs = [
+        current_app.config["AUTHOR_MATCHER_NAME_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_NAME_INITIALS_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_COLLABORATION_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_AFFILIATION_CONFIG"],
+    ]
+
+    parsed_name = ParsedName.loads(author.get("full_name"))
+    author_matcher_data = {
+        "first_name": parsed_name.first,
+        "last_name": parsed_name.last,
+        "full_name": author.get("full_name"),
+        "collaborations": get_value(record, "collaborations.value", []),
+        "affiliations": get_value(author, "affiliations.value", []),
+    }
+
+    for config in configs:
+        matched_reference = match_literature_author_with_config(
+            author_matcher_data, config
+        )
+        updated_author_recid = assign_reference_to_author(matched_reference, author)
+        if updated_author_recid:
+            updated_authors.append(updated_author_recid)
+            return author
+
+
 @shared_task(ignore_result=False, bind=True)
 def disambiguate_authors(self, record_uuid):
     record = InspireRecord.get_record(record_uuid)
     if "Literature" not in record["_collections"]:
         return
     authors = record.get_modified_authors()
     updated_authors = []
+
     for author in authors:
         if author.get("curated_relation"):
             continue
-        matched_authors = [
-            matched_record["_source"]["control_number"]
+        matched_authors = {
+            matched_record["_source"]["self"]["$ref"]
             for matched_record in match(
                 author, current_app.config["AUTHOR_MATCHER_EXACT_CONFIG"]
             )
-        ]
-        if len(set(matched_authors)) == 1:
-            author_control_number = matched_authors[0]
-            author["record"] = get_ref_from_pid("aut", author_control_number)
-            updated_authors.append(str(author_control_number))
+        }
+        updated_author_recid = assign_reference_to_author(matched_authors, author)
+        updated_authors.append(updated_author_recid)
```

Comment:
I still think it's not nice that you're mixing different levels of concern in this function (high-level: calling `match_literature_author`, low-level: the stuff you're doing here with calling `match` with an explicit config, getting the `$ref` out of it, etc.). So I think it's better to split this part into a `match_literature_author_ids` function.